cli/azd/pkg/project/dotnet_importer.go

```diff
@@ -323,7 +323,26 @@ func (ai *DotNetImporter) Services(
 
 		if bContainer.Build != nil {
 			defaultLanguage = ServiceLanguageDocker
-			relPath, err := filepath.Rel(p.Path, filepath.Dir(bContainer.Build.Dockerfile))
+			// dockerfiles are copied to the infra folder like bicep files to ensure
+			// provision and deploy works after infra gen.
+			bContainer.Build.Dockerfile = filepath.Join(
+				svcConfig.Project.Path, svcConfig.Project.Infra.Path, name, filepath.Base(bContainer.Build.Dockerfile))
```

Comment:
Requirement question: Is `azd infra gen` taking ownership of the inline Dockerfile the thing we'd want to support?